core/trino-main/src/main/java/io/trino/sql/planner/SymbolAllocator.java

```diff
@@ -88,7 +93,7 @@ public Symbol newSymbol(String nameHint, Type type, @Nullable String suffix)
         String unique = nameHint;
 
         if (suffix != null) {
-            unique = unique + "$" + suffix;
+            unique = unique + "_" + suffix;
```

Comment:
`$` was readable

but if you want to replace with `_`, do you need to update the `nameHint.lastIndexOf("_")` logic and the likes?

Reply:
Updated again with a few more changes that should address this.